src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilterProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class SaveTempDataPropertyFilterProvider : IFilterFactory
+    {
+        internal IList<PropertyHelper> TempDataProperties { get; set; }
+
+        public bool IsReusable => false;
```

Comment:
Good point about reusable. Discussed with @rynowak about the registration, it should be transient